docs/painless/painless-operators-boolean.asciidoc

```diff
@@ -0,0 +1,516 @@
+
+*Grammar:*
+[source,ANTLR4]
+----
+boolean_not: '!' expression;
+----
+
+Note that def types will be assumed to be of the boolean type.  Any def type evaluated at run-time that does not represent a boolean will result in an error.  Non-boolean expressions will result in an error.
+
+*Examples:*
+[source,Java]
+----
+boolean x = !false; // declares the boolean variable x and sets it to the opposite of the false value
+boolean y = !x;     // declares the boolean variable y and sets it to the opposite of the boolean variable x
+def z = !y;         // declares the def variable z and sets it to the opposite of the boolean variable y
+----
+
+[[greater-than-operator]]
+==== Greater Than
+
+Greater than compares two numerical expressions where a resultant boolean value will be true if the left-side expression is a larger value than the right-side expression otherwise false.  The format is an expression, followed by the right angle operator, and a closing expression.
+
+*Grammar:*
+[source,ANTLR4]
+----
+greater_than: expression '>' expression;
+----
+
+A numeric promotion may occur during a greater than operation.  A def type evaluated at run-time will follow the same promotion table at run-time following whatever type def represents. Non-numeric expressions will result in an error.
+
+Promotion Table:
+|====
+||byte|short|char|int|long|float|double|def
+|byte|int|int|int|int|long|float|double|def
+|short|int|int|int|int|long|float|double|def
+|char|int|int|int|int|long|float|double|def
+|int|int|int|int|int|long|float|double|def
+|long|long|long|long|long|long|float|double|def
+|float|float|float|float|float|float|float|double|def
+|double|double|double|double|double|double|double|double|def
+|def|def|def|def|def|def|def|def|def
+|====
+
+*Examples:*
+[source,Java]
+----
+boolean x = 5 > 4; // declares the int variable x and sets it to the result of 5 greater than 4
+double y = 7.0;    // declares the double variable y and sets it to the double constant 7.0
+def z = y > 6.5;   // declares the def variable z and sets it to the result of y greater than 6.5
+def a = y > x;     // declares the def variable a and sets it to the result of y greater than z (x is promoted to double at compile-time)
+----
+
+[[greater-than-or-equal-operator]]
+==== Greater Than Or Equal
+
+Greater than or equal compares two numerical expressions where a resultant boolean value will be true if the left-side expression is a larger value than or equal to the right-side expression otherwise false.  The format is an expression, followed by the right angle and equals operator, and a closing expression.
+
+*Grammar:*
+[source,ANTLR4]
+----
+greater_than_or_equal: expression '>=' expression;
+----
+
+A numeric promotion may occur during a greater than or equal operation.  A def type evaluated at run-time will follow the same promotion table at run-time following whatever type def represents. Non-numeric expressions will result in an error.
+
```

Comment:
Using the greater than or equal operator with non-numeric expressions results in an error. (Same for following sections)

Reply:
Changed.